features/roomlist/impl/src/test/kotlin/io/element/android/features/roomlist/impl/filters/RoomListFiltersViewTests.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.roomlist.impl.filters
+
+import androidx.activity.ComponentActivity
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.element.android.features.roomlist.impl.R
+import io.element.android.libraries.testtags.TestTags
+import io.element.android.tests.testutils.EventsRecorder
+import io.element.android.tests.testutils.clickOn
+import io.element.android.tests.testutils.pressTag
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toImmutableList
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class RoomListFiltersViewTests {
+    @get:Rule val rule = createAndroidComposeRule<ComponentActivity>()
+
+    @Test
+    fun `clicking on filters generates expected Event`() {
+        val eventsRecorder = EventsRecorder<RoomListFiltersEvents>()
+        rule.setContent {
+            RoomListFiltersView(
+                state = aRoomListFiltersState(eventSink = eventsRecorder),
+            )
+        }
+        rule.clickOn(R.string.screen_roomlist_filter_rooms)
+        eventsRecorder.assertList(
+            listOf(
+                RoomListFiltersEvents.ToggleFilter(RoomListFilter.Rooms),
+            )
+        )
+    }
+
+    @Test
+    fun `clicking on clear filters generates expected Event`() {
+        val eventsRecorder = EventsRecorder<RoomListFiltersEvents>()
+        rule.setContent {
+            RoomListFiltersView(
+                state = aRoomListFiltersState(
+                    unselectedFilters = persistentListOf(),
+                    selectedFilters = RoomListFilter.entries.toImmutableList(),
+                    eventSink = eventsRecorder
+                ),
+            )
+        }
+        rule.pressTag(TestTags.homeScreenClearFilters.value)
+        eventsRecorder.assertList(
```

Comment:
Neat: `assertSingle` is more compact than `assertList`, maybe less extensible, but I do not think the expected result will change in the future.